src/content/api/parser.md

```diff
@@ -37,279 +38,477 @@ as such:
 
 `SyncBailHook`
 
-Evaluate the type of an identifier.
+Triggered when evaluating an expression consisting in a `typeof` of a free variable
 
-Parameters: `expression`
+- Hook Parameters: `identifier`
+
+- Callback Parameters: `expression`
+
+```js
+parser.hooks.evaluateTypeof.for('myIdentifier').tap('MyPlugin', expression => {
+  /* ... */
+  return expressionResult;
+});
+```
+
+```js
+// This will trigger the evaluateTypeof hook
+const a = typeof myIdentifier;
+```
+
+```js
+// This won't
+const myIdentifier = 0;
+const b = typeof myIdentifier;
+```
 
 
 ### evaluate
 
 `SyncBailHook`
 
-Evaluate an expression.
+Called when evaluating an expression.
 
-Parameters: `expression`
+- Hook parameters: `expressionType`
+
+- Callback parameters: `expression`
+
+```js
+parser.hooks.evaluate.for(/* expression type */).tap(/* ... */);
```

Comment:
Please add strings, function examples as you did in previous hook:
```
parser.hooks.evaluateTypeof.for('myIdentifier').tap('MyPlugin', expression => {
  /* ... */
  return expressionResult;
});
```
To be consistent with providing the examples to the reader